src/lib/Hydra/Controller/API.pm

```diff
@@ -285,6 +285,23 @@ sub push_github : Chained('api') PathPart('push-github') Args(0) {
     $c->response->body("");
 }
 
+sub push_gitea : Chained('api') PathPart('push-gitea') Args(0) {
+    my ($self, $c) = @_;
+
+    $c->{stash}->{json}->{jobsetsTriggered} = [];
+
+    my $in = $c->request->{data};
+    my $url = $in->{repository}->{clone_url} or die;
+    $url =~ s/.git$//;
```

Comment:
```suggestion
    $url =~ s/.git$//;
    $url =~ s/^https?:\/\///;
```

Disclaimer: this is probably the 10th or so line of perl I've ever written :see_no_evil:

Reply:
We can follow-up for this part.

Reply:
@icewind1991 how did you apply your change? Do you use a custom fork of hydra?